2019-08-03-q-and-as-about-nix-for-haskellers.md

```diff
@@ -0,0 +1,692 @@
+    often convenient to generate a Haskell package set for use in Nix.
+
+    There are two main ways people generally generate these Haskell package sets.
+
+    One is by using the output of the `cabal` solver and translating the build
+    plan to a set of Nix derivations.  The other is by using a Stackage LTS
+    package set and translating it to a set of Nix derivations.
+
+    Generating your own Nix Haskell package sets can make sure that you are
+    getting the exact same versions of your Haskell dependencies whether you're
+    building with `nix-build`, plain `cabal`, `cabal` + Nix, plain `stack`, or
+    `stack` + Nix.  This can make Nix easy to use on teams where some people
+    prefer `cabal` (without involving Nix at all), some people prefer `stack`
+    with Nix integration, and some people want to run `nix-build` on CI, etc.
+
+    There are tools for generating a set of Nix derivations from either the
+    output of the `cabal` solver or a Stackage package set.  The two most
+    well-known tools are
+    [`haskell.nix`](https://github.com/input-output-hk/haskell.nix)
+    and [`stack2nix2`](https://github.com/input-output-hk/stack2nix).
+
+    I wrote up a short
+    [comparison](https://discourse.nixos.org/t/backport-ghc-8-6-5-to-19-03/3040/7?u=cdepillabout)
+    of these two tools on the offical
+    [Nix Discourse](https://discourse.nixos.org).
+
+    Generating your own Nix Haskell package set is often overkill if you're
+    only writing a small library, but it works very well if you're on a large
+    team or working on a big project.
+
+### 4. Minimum required configuration for Haskell with Nix
+
+_What is the minimum required configuration for a Haskell project to be built
+with nix?_
+
+This is a difficult question to answer.  The downside of the flexibility of Nix
+is that it provides many different ways to integrate with Haskell projects.
+
+The minimum required configuration changes depending on whether you want to
+build purely with `nix-build`, you want to provide a `cabal` + Nix development
+environment, you want to allow building with `stack`, etc.  It also depends on
+whether you want to use the Haskell package set defined in nixpkgs, or you want to
+generate your own Haskell package set (as described in the previous question).
+
+I've setup a
+[repo that showcases](https://github.com/cdepillabout/nix-cabal-example-project)
+a minimum required configuration for building a non-trivial Haskell project
+with both `nix-build` and `cabal` + Nix.  It uses the nixpkgs Haskell package
+set.  The README describes how it can be used.
+
+If you want to build with both `nix-build` and `cabal` + Nix, but use a package
+set generated from the `cabal` solver, you can use
+[`haskell.nix`](https://github.com/input-output-hk/haskell.nix).
+
+If you want to build with `nix-build`, `cabal` + Nix, and `stack`, while using
+a package set based on a Stackage LTS release, you should also be able to use
+`haskell.nix`.
```

Comment:
I heard about `shellFor` function. And the fact that it's really helpful. It looks common to me (some people use it) but I don't understand its purpose... Here is the example:

* https://github.com/qnikst/qnikst.github.com/blob/raw/shell.nix